playbooks/roles/deploy-osh/tasks/kubectl-install.yml

```diff
@@ -84,6 +84,9 @@
   command: kubectl cluster-info
   changed_when: false
   register: _kubectlclusterinfo
+  retries: 10
+  until: _kubectlclusterinfo is succeeded
+  delay: 5
```

Comment:
Rather than waiting for another process to bring up the cluster we should move everything that is dependent on the cluster (such as this) to another stage in the jenkins pipeline that can depend on the previous task being complete.

Reply:
It technically is...  The deploy osh only runs after the caasp enrollment.

We still need to keep that code there to prevent the deployment if cluster-info is not in a good state (when we'll pass that code to customers).

However, I believe we should also have a success condition of the deployment of the caasp cluster, and caasp enrollment. It can be set to this for example,or the ability to run other kubectl commands. But that's outside the scope of this PR imo.

Reply:
That said, I agree that we shouldn't hide issues here, we should assume a working state, and just test the state. If it fails, it means the cluster don't have the requirements for deploying OSH.